index.js

```diff
@@ -55,20 +55,29 @@ module.exports = function(search, replacement, options) {
         callback(null, file);
       }
 
-      if (options && options.skipBinary) {
-        istextorbinary.isText(file.path, file.contents, function(err, result) {
-          if (err) {
-            return callback(err, file);
+      if (options) {
+        if (options.passFilename && typeof replacement === 'function') {
+          var userReplacement = replacement;
+          replacement = function (match) {
```

Comment:
I wanted to deal with this issue by introducing new option `passFilename` but you're right. The regexp replace mode will be much more useless without passing all original arguments to the replacement callbacks.